book/src/the-database/moo-value-types.md

```diff
@@ -90,42 +90,33 @@ Object numbers are always integers.
 ```
 
-## WAIF Type
-
-_WAIFs_ are lightweight objects. A WAIF is a value which you can store in a property or a variable or inside a LIST or another WAIF. A WAIF is smaller in size (measured in bytes) than a regular object, and it is faster to create and destroy. It is also reference counted, which means it is destroyed automatically when it is no longer in use. An empty WAIF is 72 bytes, empty list is 64 bytes. A WAIF will always be 8 bytes larger than a LIST (on 64bit, 4 bytes on 32bit) with the same values in it.
-
-> Note: WAIFs are not truly objects and don't really function like one. You can't manipulate a WAIF without basically recreating a normal object (and then what's the point?). It may be better to think of a WAIF as another data type. It's closer to being a list than it is to being an object. But that's semantics, really.
+But the only mandatory element is the delegate.
 
-WAIFs are smaller than typical objects, and faster to create. A WAIF has two builtin OBJ properties, .class and .owner. A WAIF is only ever going to be 4 bytes larger than a LIST with the same values.
+Examples:
 
-OBJs grow by value_bytes(value) - value_bytes(0) for every property you set (that is, every property which becomes non-clear and takes on its own value distinct from the parent). LISTs and WAIFs both grow by value_bytes(value) for each new list element (in a LIST) or each property you set (in a WAIF). So a WAIF is never more than 4 bytes larger than a LIST which holds the same values, except WAIFs give each value a name (property name) but LISTs only give them numbers.
+```
+< $div_tag, [ class -> "background_div" ], { child_node_a, child_node_b } >
+< #54 >
+< $maze_node, [ description -> "You are in a maze of twisty passages, all alike", exits = { "north", "up" } >
+```
 
-Essentially you should consider a WAIF as something you can make thousands of in a verb without a second thought. You might make a mailing list with 1000 messages, each a WAIF (instead of a LIST) but you most likely wouldn't use 1000 objects.
+During verb execution, `this`, and `caller` can be flyweights. `player` cannot.
 
-You create and destroy OBJs explicitly with the builtins create() and recycle() (or allocate them from a pool using verbs in the core). They stay around no matter what you do until you destroy them.
+flyweights are useful for representing large quantities of small, lightweight things (like a maze or a large quantity of small items). 
 
-All of the other types you use in MOO (that require allocated memory) are reference counted. However you create them, they stay around as long as you keep them in a property or a variable somewhere, and when they are no longer used, they silently disappear, and you can't get them back.
+In particular, they have been especially structured to be able to mimic tree structured data like an HTML or XML document. There are builtins (`to_xml` and `xml_parse`) for converting properly structured flyweights to/from such documents.
```

Comment:
My brain snagged on the first sentence. Here's an attempt to un-snag it:

The syntax is specifically designed to easily express tree structured data like an HTML or XML document.